layers/+tools/shell/funcs.el

```diff
@@ -208,3 +208,12 @@ is achieved by adding the relevant text properties."
   (interactive)
   (switch-to-buffer "*shell*")
   (shell "*shell*"))
+
+;; https://stackoverflow.com/questions/6837511/automatically-disable-a-global-minor-mode-for-a-specific-major-mode
+(defun spacemacs//inhibit-global-centered-cursor-mode ()
+  "Counter-act `global-centered-cursor-mode'."
```

Comment:
If possible, better to hook locally into `centered-cursor-mode-hook`.

Reply:
Also If possible, better to get upstream to use a smarter TURN-ON argument in call to `define-global-minor-mode'` Usually a `centered-cursor-mode-on` function, that checks if the current major-mode should have the minor-mode enabled (e.g. by consulting a blacklist/whitelist of major-modes) before activating, is used as the TURN-ON argument. For an example see `global-company-mode`, `company-mode-on` and `company-global-modes`.

Note these upstream changes are good as future plans, and they are not a requirement for this PR. We will merge it even without them.

Reply:
Your approach would require to disable a minor mode from its own hook, which when I tried yielded an infinite recursion. Disabling from the global minor mode hook did not work either, surely because of the order in which hooks are run.

If you take a look at the Stackoverflow question in the comment, they tried to disable it from the incriminated major modes and it did not work because globalized minor modes use the `after-major-mode-change-hook`, which is run _after_ the major mode's hook. This is why what the guy did first, which was local into `centered-cursor-mode`, was not working.

Funny thing, they were even discussing precisely this mode, `centered-cursor-mode`.

All in all, the answer is clear, detailed, references well its sources and _works_, so I think we should stick to it.

Reply:
Ok, lets keep it as is, even though I wonder if they tried:
```elisp
(lambda ()
  (when centered-cursor-mode
    (centered-cursor-mode 0)))
```